src/async_impl/response.rs

```diff
@@ -255,8 +309,67 @@ impl Response {
     /// # Ok(())
     /// # }
     /// ```
-    pub async fn bytes(self) -> crate::Result<Bytes> {
-        hyper::body::to_bytes(self.res.into_body()).await
+    pub async fn bytes(mut self) -> crate::Result<Bytes> {
+        match &self.response_config.lowest_allowed_speed {
+            None => hyper::body::to_bytes(self.res.into_body()).await,
+            Some(lowest_speed_config) => {
+                let mut window_start_time = Instant::now();
+                let mut bytes_recieved_during_window = 0;
+                // HELP: With capacity?
+                let mut bytes_buffer = Vec::new();
+
+                while let Some(bytes) = self.res.body_mut().try_next().await? {
+                    let current_time = Instant::now();
+                    let elapsed = current_time - window_start_time;
+
+                    if elapsed > lowest_speed_config.duration_window {
+                        /*
+                            * Protection against division by zero is guranteed by requiring
+                            `duration_window` to be greater or equal to one second. `elapsed`
+                            will in other words always be > 1 second whenever this branch
+                            is executed.
+
+                            * Using '<=' over '<' is required to handle the case when
+                            lowest_speed_config.bytes_per_second is set to 0.
+
+                            * Using integers removes the need to protect against NaN, +-Infinity
+                            edge-cases, normally at the cost of precision. False positives
+                            should, however, still not be able to occur given that the
+                            if statement can equivalently be expressed as:
+
+                            "abort if bytes_recieved_during_window <= min_bytes_per_per_window"
+
+                            This equivalence leaves no room for rounding errors so long as
+                            my math remains correct:
+
+                            bytes_recieved_during_window / elapsed <= min_bytes_per_second <==>
+                            bytes_recieved_during_window / elapsed <= min_bytes_per_window / duration_window
+
+                            The surrounding `elapsed > duration_window` if condition,
```

Comment:
Realized that this doesn't necessarily hold true given that `elapsed` is floored when in `elapsed.as_secs() as usize`. Will fix later today.